### DISCUSSION TITLE
I ❤️ it!

### DISCUSSION THREAD

**Discussion by jason-rietzke** [martinwoodward/octolamp#17] [Show and tell]
I actually built the octolamp a while ago, but I only recently found the time to develop the script I've been dreaming of.
The lamp should reflect the activity in the GitHub organization, and this is the outcome:

https://user-images.githubusercontent.com/28512517/228961349-161acda6-6622-4b99-b8cc-58a4ff1a9a6f.mp4

Now the lamp is idle by default, and when it receives webhook events from GitHub, it can show a green "commits" animation (as seen in the video), a "red alert" for failed workflows, or a star animation when collecting a star on any org repo.

I even put an NFC tag on the back and created a shortcut on my smartphone to toggle the octolamp by calling the web server running on the microcontroller. This is much more convenient than constantly plugging and unplugging the lamp.

I love this lamp and am very glad @martinwoodward published this repo!

P.S: I actually built two of them ❤️ one for the office and one for myself.

**Comment by martinwoodward** [comment]
Nice!!  How have you connected the webhook from GitHub to your lamp?  Do you have something like homeassistant in the middle or is your lamp exposed on the internet for GitHub to ping?

**Reply by jason-rietzke** [reply]
Thanks 😊 There is actually a small microservice running on one of my servers that uses the webhooks and from which the lamp retrieves the latest data. (I might update this to a websocket connection at some point).
This makes it pretty easy to connect any number of lamps to a single webhook.

**Reply by martinwoodward** [reply]
Makes sense, and sounds like a great solution.  If you post that code feel free to share a link here as I'm sure others would love to do a similar set up.

**Reply by jason-rietzke** [reply]
I think I might add a few features before but afterwards I will probably publish and post it here 😄

**Comment by Beatrix-droid** [comment]
@jason-rietzke hi, I love your lamp build! I've built one myself, and I too was interested in using the WLED's APIs to trigger some colour changes in my lamp based on a condition. But I've had some difficulties with it. Would you be able to provide me with a very very brief  rough outline on how you did it? I know I have to write a script that interacts both with the GitHub api and,  based on a condition , interacts with the wled one, but I am not sure where to host this script, (computer vs the chip itself , vs the phone app or somewhere in the cloud) and as such am unsure in which language to write it in.

**Reply by jason-rietzke** [reply]
Hey, @Beatrix-droid glad you like it :) Sure I can give you a brief overview of what I chose to do, but I actually forgot to publish the repo and just changed that, feel free to [take a look](https://github.com/jason-rietzke/octo-lamp).

Generally, I'm running a service (via docker) on a server and created a GitHub Webhook to send certain events to it. In my case of an entire organization but this also works for repositories I think.

I used a `Wemos D1 Mini` to run the Arduino script which just simply pulls new results from the server and plays some pre-programmed animations based on that. The Wemos also serves a website when visited via the IP which exposes a few controls – turn on / turn off; static animations; etc.
